lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/files/JobJarMaker.java

```diff
@@ -96,42 +103,87 @@ public File createJobSubmissionJAR(
       final JobSubmissionEvent jobSubmissionEvent,
       final Configuration driverConfiguration) throws IOException {
 
-    // Copy all files to a local job submission folder
-    final File jobSubmissionFolder = makejobSubmissionFolder();
-    LOG.log(Level.FINE, "Staging submission in {0}", jobSubmissionFolder);
+    return new JarBuilder()
+        .withConfiguration(driverConfiguration)
+        .withGlobalFileSet(jobSubmissionEvent.getGlobalFileSet())
+        .withLocalFileSet(jobSubmissionEvent.getLocalFileSet())
+        .withConfigurationFileName(this.fileNames.getDriverConfigurationName())
+        .build();
+  }
 
-    final File localFolder = new File(jobSubmissionFolder, this.fileNames.getLocalFolderName());
-    final File globalFolder = new File(jobSubmissionFolder, this.fileNames.getGlobalFolderName());
+  private File makejobSubmissionFolder() throws IOException {
+    return Files.createTempDirectory(this.fileNames.getJobFolderPrefix()).toFile();
+  }
 
-    copy(jobSubmissionEvent.getGlobalFileSet(), globalFolder);
-    copy(jobSubmissionEvent.getLocalFileSet(), localFolder);
+  /**
+   * Builder class for building JAR files.
+   */
+  public final class JarBuilder {
```

Comment:
Does this have to be `public`?

Reply:
Yes. AzureBatchEvaluatorShimManager.java needs to call withGlobalFileSet(), build(), withLocalFileSet() methods defined in this class.